lib/conjur/puppet_module/config.rb

```diff
@@ -27,11 +27,20 @@ def from_registry
             unless Puppet.features.microsoft_windows?
 
           require 'win32/registry'
-          Win32::Registry::HKEY_LOCAL_MACHINE.open(REG_KEY_NAME) do |reg|
-            # Convert registry value names from camel case to underscores
-            # e.g. ApplianceUrl => appliance_url
-            reg.map { |name, _type, data|  [name.gsub(/(.)([A-Z])/, '\1_\2').downcase, data] }.to_h
+
+          c = {}
+          begin
+            Win32::Registry::HKEY_LOCAL_MACHINE.open(REG_KEY_NAME) do |reg|
+              # Convert registry value names from camel case to underscores
+              # e.g. ApplianceUrl => appliance_url
+              c = reg.map { |name, _type, data|  [name.gsub(/(.)([A-Z])/, '\1_\2').downcase, data] }.to_h
+            end
+          rescue
+            Puppet.notice "Agent's registry did not contain path #{REG_KEY_NAME}. This is normal " +
```

Comment:
I'm not sure this is always normal e.g. when this error is thrown when using HFTs for the (n+1)-th time

Reply:
Sure - that's why the disclaimer is there for "when using HFTs for the first time on a node" on the second line. I'm not sure we can improve this too much since it's an edge case where this condition is not a very suspicious state on the node.